website/content/docs/api-clients/api.mdx

```diff
@@ -79,3 +81,184 @@ The following method conventions are used within Boundary's API:
+Each response contains the following headers:
+
+- `RateLimit` - Provides the current limit, number of remaining requests, and the time at which the quota will reset for the limit that is closest to being exhausted for the requested resource and action.
+- `RateLimit-Policy` - Describes the limits for the requested resource and action.
+
+If the request is limited, Boundary sends the client a 429 HTTP status code with a `Retry-After` header.
+The `Retry-After` header contains the number of seconds the client should wait before it sends the request again.
+
+Refer to the [controller stanza](/boundary/docs/configuration/controller#api_rate_limit) documentation for the specific `api_rate_limit` configuration options.
+Some example configurations are listed below.
+
+### Rate limiting configuration examples
+
+The following example shows a simple configuration where the same limits are applied to all resources and examples:
+
+```hcl
+controller {
+  # total limit for all resources and actions
+  api_rate_limit {
+	resources = ["*"]
+	actions   = ["*"]
+	per  	= "total"
+	limit 	= 500
+	period	= "1s"
+  }
+
+  # limit for ip addresses to all resources+actions to prevent a malicious
+  # host that is fabricating tokens, or spamming unauthed endpoints
+  api_rate_limit {
+	resources = ["*"]
+	actions   = ["*"]
+	per 	= "ip-address"
+	limit 	= 100
+	period	= "1s"
+  }
+
+  # Limit of all authed requests, to prevent one user consuming all of the total limit
+  api_rate_limit {
+	resources = ["*"]
+	actions   = ["*"]
+	per 	= "auth-token"
+	limit 	= 100
+	period	= "1s"
+  }
+
+  # other controller configuration
+}
+```
+
+The following example shows a configuration that disables rate limiting.
+You may want to disable rate limiting if you already use an external system like a reverse proxy to apply rate limiting:
+
+```hcl
+controller {
+  api_rate_limit_disable = true
+
+  # other controller configuration
+}
+```
+
+The following example uses the default settings for most endpoints, but configures a single override:
+
+```hcl
+controller {
+  api_rate_limit {
+	resources = ["target"]
+	actions   = ["list"]
+	per 	= "auth-token"
+	limit 	= 10
+	period	= "1s"
+  }
+
+  # other controller configuration
+}
+```
+
+The following example is more complex.
+Initially it sets some defaults to apply to all resources and actions.
+Then it configures some specific endpoints with different limits.
+
+```hcl
+controller {
+  # total limit for all resources and actions
+  api_rate_limit {
+	resources = ["*"]
+	actions   = ["*"]
+	per 	= "total"
+	limit 	= 500
+	period	= "1s"
+  }
+
+  # total limit for all list operations
+  # sets an overall cap on list since it is relatively expensive.
+  api_rate_limit {
+	resources = ["*"]
+	actions   = ["list"]
+	per 	= "total"
+	limit 	= 200
+	period	= "1s"
+  }
+
+  # Limit for ip addresses to all resources+actions to prevent a malicious
+  # host that is fabricating tokens, or spamming unauthed endpoints
```

Comment:
I would probably drop these comments from the examples. I am not sure they add anything meaningful.

Reply:
I missed some of these comments on the first pass, but just went back and cleaned them up in the latest commit.

Reply:
I just had a chat with @stellarsquall and I think we’d like to add these comments back. They were useful for us when we were writing the doc, and it’s consistent with how we present example code blocks elsewhere in the docs. I apologize for the multiple edits.